frontend/src/component/ui/Table.spec.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { Table } from './Table';
+
+describe('Table component', () => {
+  const sampleHeaders = ['Header1', 'Header2', 'Header3'];
+  const sampleRows = [
+    ['row1-cell1', 'row1-cell2', 'row1-cell3'],
+    ['row2-cell1', 'row2-cell2', 'row2-cell3'],
+    ['row3-cell1', 'row3-cell2', 'row3-cell3']
+  ];
+
+  test('should render without crash', () => {
+    render(<Table />);
+  });
+
+  test('should have no children for thead when Headers prop not passed', () => {
+    const toastRef = React.createRef<Table>();
+    const { container } = render(<Table ref={toastRef} rows={sampleRows} />);
+
+    expect(container.querySelector('thead')!.childElementCount).toBe(0);
```

Comment:
I think we should first check `container.querySelector('thead')` is not null. Those test should fail correctly because the logic changes when people are refactoring the table component. So did the rest of the tests.